slam_toolbox/solvers/ceres_solver.cpp

```diff
@@ -37,7 +38,9 @@ CeresSolver::CeresSolver() :
   first_node_ = nodes_->end();
 
   // formulate problem
-  angle_local_parameterization_ = AngleLocalParameterization::Create();
+  local_parameterization_ = new ceres::ProductParameterization(
```

Comment:
Please explain this

Reply:
I change the formulation of the NLS problem to optimize SE3 variables instead of 3 scalar variables so as to maintain a 1-to-1 relationship between nodes and parameter blocks. That allows me to reconstruct block ordering to make sense of the information matrix.

Reply:
https://github.com/ceres-solver/ceres-solver/blob/b0aef211db734379319c19c030e734d6e23436b0/docs/source/nnls_modeling.rst#classproductparameterization I noticed there is a depreciation warning, should we be using something else instead?

Reply:
Hmm, I had not seen that. It looks like `Parameterization` abstractions have been deprecated in [Ceres 2.1.0rc1](https://github.com/ceres-solver/ceres-solver/blob/2.1.0rc1/docs/source/nnls_modeling.rst#classproductparameterization), but as far as I can see that version is not yet available in any Linux distro. Certainly not for Debian/Ubuntu, and [by the looks of it](https://tracker.debian.org/pkg/ceres-solver) I don't think it'll make it to Jammy.

We should probably keep an eye on it though, we may have to transition in the next couple years.